client/src/modules/Sidebar/components/EditForm/index.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable camelcase */
+import React, { useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import MyForm from '../../../../shared/Form';
+import { editItem,fetchSingleDefById } from '../../api';
+import Loader from '../../../../shared/Loader';
+import cancelToken from '../../../../shared/cancel-token';
+import { setMapCenter } from '../../../MapHolder/actions/mapState';
+import { getReverseGeocoding } from '../../../../shared/api';
+
+const defCancelToken = cancelToken();
+
+const useStyles = makeStyles({
+  loader: {
+    padding: '44%'
+  }
+});
+
+const EditForm = ({ setMapCenter }) => {
+  const [def, setDef] = useState(null);
+
+  const classes = useStyles();
+  const { href } = window.location;
+  const id = href.slice(href.lastIndexOf('/') + 1);
+
+  const prepareData = async ({
+    data: { defibrillator }
+  }) => {
+    const [lng, lat] = defibrillator.location.coordinates;
+    const correctAddress = await getReverseGeocoding({
+      lng,
+      lat
+    });
+
+    setDef({
+      title: defibrillator.title,
+      address:
+        correctAddress.data.results[0].formatted_address,
+      informational_plates:
+        defibrillator.informational_plates,
+      phone: defibrillator.phone,
+      additional_information:
+        defibrillator.additional_information,
+      floor: defibrillator.storage_place.match(/\d/)[0],
+      storage_place: defibrillator.storage_place.match(
+        /, (.*)/
+      )[1],
+      coordinates: defibrillator.location.coordinates,
+      accessibility: defibrillator.accessibility
+    });
+    setMapCenter({ lng, lat, zoom: 17 });
+  };
+
+  useEffect(() => {
+    (async () => {
+      setDef(null);
+      const res = await fetchSingleDefById(id);
+      prepareData(res);
+    })();
+    return () => {
+      defCancelToken.cancel();
+    };
+  }, []);
+
+  const hadleSubmit = async ({ images, ...data }) => {
+    const body = {
+      ...data,
+      _id: id,
+      actual_date: data.actualDate,
+      location: {
+        type: 'Point',
+        coordinates: data.coordinates
+      },
+      storage_place: `Поверх ${data.floor}, ${data.storage_place}`,
+      language: 'Україномовний'
```

Comment:
This is hard coded property. If current def has another language interface it will be owerwritten.